In [2]:
import gzip
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt





In [3]:
#read zip file into a dataframe
movie_gross_df = pd.read_csv('zippedData/bom.movie_gross.csv.gz')
movie_gross_df.head() #preview file


title studio  domestic_gross  \
0                                  Toy Story 3     BV     415000000.0   
1                   Alice in Wonderland (2010)     BV     334200000.0   
2  Harry Potter and the Deathly Hallows Part 1     WB     296000000.0   
3                                    Inception     WB     292600000.0   
4                          Shrek Forever After   P/DW     238700000.0   

  foreign_gross  year  
0     652000000  2010  
1     691300000  2010  
2     664300000  2010  
3     535700000  2010  
4     513900000  2010

In [6]:
print(movie_gross_df.shape) #5 columns with 3387 records

(3387, 5)


In [7]:
movie_gross_df.columns

Index(['title', 'studio', 'domestic_gross', 'foreign_gross', 'year'], dtype='object')

In [8]:
movie_gross_df.isnull().sum()

title                0
studio               5
domestic_gross      28
foreign_gross     1350
year                 0
dtype: int64

In this data there are 5 null studio values, 28 null domestic gross values, and the foerign_gross has 1350 null values out of a possible 3387 records.


In [11]:
movie_gross_df.title.value_counts()

Bluebeard                         2
Beyond the Reach                  1
Rare Exports: A Christmas Tale    1
Another Happy Day                 1
Remember You                      1
                                 ..
21 and Over                       1
Shoplifters                       1
Finding Dory                      1
Elliot: The Littlest Reindeer     1
Frozen                            1
Name: title, Length: 3386, dtype: int64

Bluebeard is the only title that appears more than once. However, I have not been able to see if there are similar, but different titles.

In [50]:
len(movie_gross_df.studio.unique()) 

movie_studios = movie_gross_df.studio.str.lower().unique()
len(movie_studios) #258 studios
print(movie_studios)

    


['bv' 'wb' 'p/dw' 'sum.' 'par.' 'uni.' 'fox' 'wein.' 'sony' 'foxs' 'sgem'
 'wb (nl)' 'lgf' 'mbox' 'cl' 'w/dim.' 'cbs' 'focus' 'mgm' 'over.' 'mira.'
 'ifc' 'cj' 'nm' 'spc' 'parv' 'gold.' 'js' 'ratt.' 'magn.' 'free' '3d'
 'utv' 'rela.' 'zeit.' 'anch.' 'pda' 'lorb.' 'app.' 'drft.' 'osci.' 'iw'
 'rog.' nan 'eros' 'relbig.' 'viv.' 'hann.' 'strand' 'nge' 'scre.' 'kino'
 'abr.' 'cz' 'ato' 'first' 'gk' 'find.' 'nfc' 'tfc' 'pala.' 'imag.' 'nav'
 'arth.' 'cls' 'mont.' 'olive' 'cgld' 'foak' 'ivp' 'yash' 'icir' 'fm'
 'vita.' 'wow' 'truly' 'indic.' 'fd' 'vari.' 'tris' 'orf' 'im' 'elev.'
 'cohen' 'neoc' 'jan.' 'mne' 'trib.' 'rocket' 'omni/fsr' 'kkm' 'argo.'
 'smod' 'libre' 'frun' 'whe' 'p4' 'kc' 'sd' 'am' 'mpft' 'icar.' 'agf'
 'a23' 'da.' 'nyer' 'rialto' 'df' 'kl' 'alp' 'lg/s' 'wgusa' 'mpi' 'rtwc'
 'fip' 'rf' 'arcent' 'paluni' 'epicpics' 'eone' 'ld' 'af' 'tfa' 'myr.'
 'bm&dh' 'seg' 'palt' 'outs' 'outf' 'bsm' 'wamcr' 'pm&e' 'a24' 'cdgm.'
 'distrib.' 'imax' 'ph' 'htr' 'els' 'pi' 'e1' 'tvc' 'fef' 'excl

#### Studio Data

there are 258 studios listed with no obvious mistakes. There may be some consolidation such as 'WB (NL)' can possibly be looked at as WB.

In [53]:
print(movie_gross_df.year.value_counts())

2015    450
2016    436
2012    400
2011    399
2014    395
2013    350
2010    328
2017    321
2018    308
Name: year, dtype: int64


#### Year data

Year data is clean and ranging from 2012-2018

In [7]:
domestic_gross_nullRecords = movie_gross_df['domestic_gross'].loc[movie_gross_df['domestic_gross'].isnull() == True]
#print(domestic_gross_nullRecords)
#len(domestic_gross_nullRecords) # 28 null records

domestic_gross_df = movie_gross_df.dropna(subset=['domestic_gross'])


#### Domestic Gross data
This data is prety clean and is showing only 28 missing values out of over 3000 records. These can/should be simply removed from the data set.

In [8]:
domestic_gross_df.isnull().sum()

title                0
studio               3
domestic_gross       0
foreign_gross     1350
year                 0
dtype: int64

In [13]:
domestic_gross_minuscolumns_df = domestic_gross_df.drop(columns= ['foreign_gross','studio'])
domestic_gross_minuscolumns_df.head()

title  domestic_gross  year
0                                  Toy Story 3     415000000.0  2010
1                   Alice in Wonderland (2010)     334200000.0  2010
2  Harry Potter and the Deathly Hallows Part 1     296000000.0  2010
3                                    Inception     292600000.0  2010
4                          Shrek Forever After     238700000.0  2010